src/jit/morph.cpp

```diff
@@ -450,14 +450,18 @@ GenTreePtr Compiler::fgMorphCast(GenTreePtr tree)
             // For these operations the lower 32 bits of the result only depends
             // upon the lower 32 bits of the operands
             //
-            if ((oper->OperGet() == GT_ADD) || (oper->OperGet() == GT_MUL) || (oper->OperGet() == GT_AND) ||
-                (oper->OperGet() == GT_OR) || (oper->OperGet() == GT_XOR))
+            if ((oper->OperGet() == GT_ADD) || (oper->OperGet() == GT_SUB) || (oper->OperGet() == GT_MUL) ||
```

Comment:
How about
```
oper->OperIs(GT_ADD, GT_SUB, GT_MUL, GT_AND, GT_OR, GT_XOR, GT_NOT, GT_NEG, GT_LSH)
```
instead of the repeated `(oper->OperGet() == GT_XYZ)`?